server/api/sectionevent/sectionevent.controller.js

```diff
@@ -62,13 +63,99 @@ function fullRemove(res){
     }
 }
 
+function withDefault(queryString, defaultValue){
+  if (queryString === undefined) return defaultValue;
+  else return queryString.toLowerCase()==="true";
+}
+
+
 // Gets a list of SectionEvents
 exports.index = function(req, res) {
   SectionEvent.findAsync()
     .then(responseWithResult(res))
     .catch(handleError(res));
 };
+// Gets a list of SectionEvents
+exports.index = function(req, res) {
+  var onlyNumber = withDefault(req.query.onlyNumber, false);
+  var withEventInfo = withDefault(req.query.withEventInfo, !onlyNumber && true);
+  var withAuthor = withDefault(req.query.withAuthor, !onlyNumber && true);
+  var withSection = withDefault(req.query.withSection, false);
+  var withCourse= withDefault(req.query.withCourse, withSection);
+
+  function respond(query){
+    if (withAuthor){
+      query.populate("author");
+    }
+    if (withEventInfo){
+      query.populate("info");
+    }
+    if (withSection){
+      if (withCourse){
+        query.populate({
+          path: 'section',
+          model: 'Section',
+          populate: {
+            path: 'course',
+            model: 'Course'
+          }
+        });
+      }else query.populate("section");
+    }
+
+    query.sort({time: -1});
 
+    query.execAsync()
+      .then(responseWithResult(res))
+      .catch(handleError(res));
+  }
+
+  if (req.query.onlyAuthor || req.query.onlySection || req.query.onlyEvent){
+    var search = {};
+    if (req.query.onlyAuthor) {
+      var authorId = req.query.onlyAuthor.toLowerCase() === "me" ? req.user._id : req.query.onlyAuthor;
+      search.author = authorId;
+    }
+    if (req.query.onlySection) search.section = req.query.onlySection;
+    if (req.query.onlyEvent) search.info = req.query.onlyEvent;
+    if (onlyNumber){
+        SectionEvent.count(search)
+        .execAsync()
+        .then(responseWithResult(res))
+        .catch(handleError(res));
+    }
+    else{
+      respond(SectionEvent.find(search));
+    }
+  }else if (req.query.onlyUserSections){
+    var user = req.query.onlyUserSections.toLowerCase()=== "me" ? req.user._id : req.query.onlyUserSections;
+    Section.findAsync({ $or: [{ instructors: user}, { students: user} ]})
+      .then((sections) => {
+        var sectionIds = sections.map((sec) => sec._id);
+        if (onlyNumber){
+            SectionEvent.count({section: {$in: sectionIds}})
+            .execAsync()
+            .then(responseWithResult(res))
+            .catch(handleError(res));
+        }
+        else{
+          respond(SectionEvent.find({section: {$in: sectionIds}}))
+        }
+      });
+  }else if (req.query.onlyNumber){
+    SectionEvent.count()
+    .execAsync()
+        .then((entity)=>{
+          console.log(entity);
+          return entity;
```

Comment:
respondWithResults sends a json object, but a single item is being passed to it, 
try returning {"number": entity} instead of just entity. 
also remove the console.log !